head/firmware/motor_hardware.c

```diff
@@ -189,6 +194,103 @@ void MX_GPIO_Init(void) {
     GPIO_InitStruct.Pull = GPIO_NOPULL;
     GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
     HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+    /*Configure GPIO pin : B1_Pin */
+    GPIO_InitStruct.Pin = GPIO_PIN_13;
+    GPIO_InitStruct.Mode = GPIO_MODE_IT_RISING;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    HAL_GPIO_Init(GPIOC, &GPIO_InitStruct);
+
+    /* EXTI interrupt init*/
+    HAL_NVIC_SetPriority(EXTI15_10_IRQn, 0, 0);
+    HAL_NVIC_EnableIRQ(EXTI15_10_IRQn);
+}
+
+void MX_TIM2_Init(void){
+
+    TIM_Encoder_InitTypeDef sConfig = {0};
+    TIM_MasterConfigTypeDef sMasterConfig = {0};
+    TIMEx_EncoderIndexConfigTypeDef sEncoderIndexConfig = {0};
+    __HAL_RCC_TIM2_CLK_ENABLE();
+    htim2.Instance = TIM2;
+    htim2.Init.Prescaler = 0;
+    htim2.Init.CounterMode = TIM_COUNTERMODE_UP;
+    htim2.Init.Period = 4.294967295E9;
```

Comment:
this is autogenerated, right? this is a very odd way to write this - i assume what it's getting at is "the largest value you can have in a 32 bit register". Instead, we can use the macro `UINT32_MAX`, defined in [stdint.h](https://en.cppreference.com/w/c/types/integer)
```suggestion
    htim2.Init.Period = UINT32_MAX;
```

Reply:
that value was autogenerated but the cubemx software. That looks better.